### ISSUE TITLE
Add support for repository pattern for different databases? 

### ISSUE THREAD

**Issue by GlennArens** [snatch-dev/Convey#1]
Hey,
Maybe there should be support for different database types with for example dapper with a generic repository pattern on top.

**Comment by GooRiOn** [comment]
Good point! I think that we should extract abstractions for persistence first (something like Convey.Persistence) and then we will think about particular implementations for SQL/NoSQL databases.

**Comment by millicandavid** [comment]
I know my group was interested in a CosmosDB and a SQL Server implementation. The CosmosDB implementation may not be necessary as the MongoDB adapter for it may work just fine. We haven't experimented with that yet though. I'll update here if we find that it does.

**Comment by dsoltesz** [comment]
Would like to see a Convey.Persistence.EFCoreDB

**Comment by GooRiOn** [comment]
Ok, I will prioritize this one to extrat the `IRepository<TEntity, TKey>` to the separated package.

**Comment by Dryadepy** [comment]
> Good point! I think that we should extract abstractions for persistence first (something like Convey.Persistence) and then we will think about particular implementations for SQL/NoSQL databases.

I'think its not necessary to make an abstractions for persistence as in fact will make it a bit hard to extend later (support for multiple nosql provides)

**Comment by frankely** [comment]
> 
> 
> > Good point! I think that we should extract abstractions for persistence first (something like Convey.Persistence) and then we will think about particular implementations for SQL/NoSQL databases.
> 
> I'think its not necessary to make an abstractions for persistence as in fact will make it a bit hard to extend later (support for multiple nosql provides)

I think it would be more flexible to choose between a SQL or NoSQL concrete NuGet package since they both have different features and trying to abstract them will reduce flexibility.
